pkg/tnf/reel/reel.go

```diff
@@ -214,6 +214,10 @@ func (r *Reel) Step(step *Step, handler Handler) error {
 				}
 				match, matchStatus := r.stripEmulatedPromptFromOutput(result.Match[0])
 				log.Debugf("command status: output=%s, match=%s, outputStatus=%d, matchStatus=%d", output, match, outputStatus, matchStatus)
+				// If the exit code is not zero, then fail the test
+				if outputStatus != 0 {
```

Comment:
We already check and set r.Err. I think what's missing is the check on r.Err at the end, outside the loop. Looks like a miss from the original design. The original implementation only checks at the beginning of the loop, so the err from the last step gets ignored.

Reply:
also, line 76-78 in test.go can be removed after fixing this?

Reply:
fixed